ecommerce_integrations/shopify/order.py

```diff
@@ -242,46 +252,67 @@ def get_tax_account_description(tax):
 	each such shipping detail consists of a list of tax_lines"""
 	for shipping_charge in shipping_lines:
 		if shipping_charge.get("price"):
-
 			shipping_discounts = shipping_charge.get("discount_allocations") or []
 			total_discount = sum(flt(discount.get("amount")) for discount in shipping_discounts)
 
 			shipping_taxes = shipping_charge.get("tax_lines") or []
-			total_tax = sum(flt(discount.get("price")) for discount in shipping_taxes)
+			total_tax = sum(flt(d.get("price")) for d in shipping_taxes)
 
 			shipping_charge_amount = flt(shipping_charge["price"]) - flt(total_discount)
 			if bool(taxes_inclusive):
 				shipping_charge_amount -= total_tax
 
-			taxes.append(
-				{
-					"charge_type": "Actual",
-					"account_head": get_tax_account_head(shipping_charge),
-					"description": get_tax_account_description(shipping_charge) or shipping_charge["title"],
-					"tax_amount": shipping_charge_amount,
-					"cost_center": setting.cost_center,
-				}
-			)
+			if cint(setting.add_shipping_as_item) and setting.shipping_item:
+				items.append(
+					{
+						"item_code": setting.shipping_item,
+						"rate": shipping_charge_amount,
+						"delivery_date": items[-1]["delivery_date"] if items else nowdate(),
+						"qty": 1,
+						"stock_uom": "Nos",
+						"warehouse": setting.warehouse,
+					}
+				)
+			else:
+				account_head = get_tax_account_head(shipping_charge)
+				if tax_account_wise_data.get(account_head):
+					tax_account_wise_data[account_head]["tax_amount"] += shipping_charge_amount
+				else:
+					tax_account_wise_data.update({
+						account_head: {
+							"charge_type": "Actual",
+							"account_head": account_head,
+							"description": get_tax_account_description(shipping_charge) or shipping_charge["title"],
+							"tax_amount": shipping_charge_amount,
+							"cost_center": setting.cost_center,
+						}
+					})
 
 		for tax in shipping_charge.get("tax_lines"):
-			taxes.append(
-				{
-					"charge_type": "Actual",
-					"account_head": get_tax_account_head(tax),
-					"description": (
-						f"{get_tax_account_description(tax) or tax.get('title')} - {tax.get('rate') * 100.0:.2f}%"
-					),
-					"tax_amount": tax["price"],
-					"cost_center": setting.cost_center,
-				}
-			)
-
-	return taxes
+			account_head = get_tax_account_head(tax)
+			if tax_account_wise_data.get(account_head):
+				tax_account_wise_data[account_head]["tax_amount"] += flt(tax["price"])
+				if cint(setting.add_shipping_as_item) and setting.shipping_item:
+					tax_account_wise_data[account_head]["item_wise_tax_detail"].update({
+						setting.shipping_item: [flt(tax.get("rate")) * 100, flt(tax.get("price"))]
+					})
+			else:
+				tax_account_wise_data.update({
+					account_head: {
+						"charge_type": "Actual",
+						"account_head": account_head,
+						"description": (
+							f"{get_tax_account_description(tax) or tax.get('title')}"
+						),
+						"tax_amount": tax["price"],
+						"cost_center": setting.cost_center,
```

Comment:
Why are you not updating `item_wise_tax_detail` here?
Also its just equal to using `set_default` instead of if / else

Reply:
Have added `item_wise_tax_detail`